pkg/pod/pod_test.go

```diff
@@ -40,41 +41,46 @@ var (
 		CredsImage:      "override-with-creds:latest",
 		ShellImage:      "busybox",
 	}
+
+	ignoreReleaseAnnotation = func(k string, v string) bool {
+		return k == ReleaseAnnotation
+	}
 )
 
 func TestMakePod(t *testing.T) {
-	names.TestingSeed()
-
-	implicitEnvVars := []corev1.EnvVar{{
-		Name:  "HOME",
-		Value: homeDir,
-	}}
-	secretsVolumeMount := corev1.VolumeMount{
-		Name:      "tekton-internal-secret-volume-multi-creds-9l9zj",
-		MountPath: "/tekton/creds-secrets/multi-creds",
-	}
-	secretsVolume := corev1.Volume{
-		Name:         "tekton-internal-secret-volume-multi-creds-9l9zj",
-		VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "multi-creds"}},
-	}
+	var (
+		secretsVolumeMount = corev1.VolumeMount{
+			Name:      "tekton-internal-secret-volume-multi-creds-9l9zj",
+			MountPath: "/tekton/creds-secrets/multi-creds",
+		}
+		secretsVolume = corev1.Volume{
+			Name:         "tekton-internal-secret-volume-multi-creds-9l9zj",
+			VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "multi-creds"}},
+		}
 
-	placeToolsInit := corev1.Container{
-		Name:         "place-tools",
-		Image:        images.EntrypointImage,
-		Command:      []string{"cp", "/ko-app/entrypoint", "/tekton/tools/entrypoint"},
-		VolumeMounts: []corev1.VolumeMount{toolsMount},
-	}
+		implicitEnvVars = []corev1.EnvVar{{
+			Name:  "HOME",
+			Value: homeDir,
+		}}
 
-	runtimeClassName := "gvisor"
-	automountServiceAccountToken := false
-	dnsPolicy := corev1.DNSNone
-	enableServiceLinks := false
-	priorityClassName := "system-cluster-critical"
+		placeToolsInit = corev1.Container{
+			Name:         "place-tools",
+			Image:        images.EntrypointImage,
+			Command:      []string{"cp", "/ko-app/entrypoint", "/tekton/tools/entrypoint"},
+			VolumeMounts: []corev1.VolumeMount{toolsMount},
+		}
+		runtimeClassName             = "gvisor"
+		automountServiceAccountToken = false
+		dnsPolicy                    = corev1.DNSNone
+		enableServiceLinks           = false
+		priorityClassName            = "system-cluster-critical"
+	)
 
 	for _, c := range []struct {
 		desc            string
 		trs             v1alpha1.TaskRunSpec
 		ts              v1alpha1.TaskSpec
+		featureFlags    map[string]string
```

Comment:
this might be a subject for a larger discussion but a couple thoughts here:
- the more feature flags we add, the larger the matrix of test cases we're going to need
- does it make sense for the `pod` package to know about feature flags, or could it be that we tell the `pod` package something more specific like a bool for "update status early if no sidecars"?

Reply:
> the more feature flags we add, the larger the matrix of test cases we're going to need
Indeed though I can't think of a good way around adding more tests at the moment! 

> does it make sense for the pod package to know about feature flags, or could it be that we tell the pod package something more specific like a bool for "update status early if no sidecars"?

I think we should refactor the feature flags bit into its own package (#2363 is somewhat related)